src/Kobo/SystemDialog.cpp

```diff
@@ -224,6 +244,33 @@ SystemWidget::UpdateBacklightButtons(int percent)
   }
 }
 
+inline void
+SystemWidget::IncreaseBacklightColourTemperature()
+{
+  int current_colour = KoboGetBacklightColour();
+  KoboSetBacklightColour(current_colour + 2);
+  UpdateBacklightColourButtons(current_colour + 2);
+}
+
+inline void
+SystemWidget::DecreaseBacklightColourTemperature()
+{
+  int current_colour = KoboGetBacklightColour();
+  KoboSetBacklightColour(current_colour - 2);
```

Comment:
What if the getter has failed? Why call the setter if it has failed?

Reply:
I don't see how the getter could fail.  The getter is now tested for presence of the file prior to enabling the buttons.

Reply:
OK.  I've fixed it to silently return if the getter fails.